controllers/dspastatus/dspa_status.go

```diff
@@ -0,0 +1,191 @@
+package dspastatus
+
+import (
+	"fmt"
+	dspav1alpha1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1alpha1"
+	"github.com/opendatahub-io/data-science-pipelines-operator/controllers/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DSPAStatus interface {
```

Comment:
> do you foresee a need for multiple DSPAStatus implementations in the future?

I don't.

By using an interface we:

- Encapsulate the instance fields
- Have one unique way to create instances (one single factory method instead of having a "constructor" and the struct initializer)
- Can initialize fields with default values
- Avoid having instances in invalid states (`nil` fields, for example)
- Can make changes more easily/less impactful, due to encapsulation and callers not depending on implementation

Reply:
While not all of these require interfaces, the points about encapsulation are valid

another reason would be is that it would be easier to mock for unit testing (we don't do this enough)